prow/cmd/phaino/local.go

```diff
@@ -550,3 +546,24 @@ func (opts *options) convertJob(ctx context.Context, log *logrus.Entry, pj prowa
 	}
 	return fmt.Errorf("grace period expired, aborted: %v", ctx.Err())
 }
+
+func getTimeout(pj prowapi.ProwJob, opts *options) time.Duration {
+	if pj.Spec.DecorationConfig.Timeout.Duration > 0 {
+		return pj.Spec.DecorationConfig.Timeout.Duration
+	}
+	if opts.timeout > 0 {
+		return opts.timeout
+	}
+	return time.Duration(0)
+}
+
+func getMinimumGracePeriod(minimum time.Duration, pj prowapi.ProwJob, opts *options, log *logrus.Entry) time.Duration {
```

Comment:
This function only needs `pj.Spec.DecorationConfig.GracePeriod.Duration` and `opts.grace` instead of `pj` and `grace`, please limit to those so that it's easier to write unit test. And speaking of unit test, please add them as well